packages/sdk/src/addresses/index.ts

```diff
@@ -103,5 +103,10 @@ export function getAddressesFromPublicKey(
   return [getAddressFromBip32PublicKey(bip32PublicKey, network, type)];
 }
 
+export function getNetworkByAddress(address: string): Network {
+  const { network: validatedNetwork } = getAddressInfo(address);
+  return validatedNetwork;
+}
```

Comment:
this will throw an error from `bitcoin-address-validation` and leak to consumers, wrap it in try-catch and `throw OrditSDKError("Invalid address")` (see `getAddressFormatForRegTest`)